traits/tests/test_has_traits.py

```diff
@@ -307,3 +307,8 @@ def _event_fired(self):
         old_count = target.event_count
         target.event = event
         self.assertEqual(target.event_count, old_count + 1)
+
+    def test_traits_inited(self):
```

Comment:
Looks like we need unit tests for `_set_traits_inited`, too. One way to test would be to override `__init__` in a `HasTraits` subclass: before a call to the superclass `__init__`, the class should be uninitialized, so you have a chance to check that `_set_traits_inited` works as advertised.